contracts/layer-2/ChampMarketplace.sol

```diff
@@ -632,39 +679,48 @@ contract ChampMarketplace is AccessControlEnumerableUpgradeable {
             isReservationOpenFor(nftReceiver, tokenId),
             "ChampMarketplace: A reservation exists for this sale"
         );
-        require(
-            _CHAMP_TOKEN_CONTRACT.allowance(msg.sender, address(this)) >=
-                salePrice,
-            "ChampMarketplace: Allowance is lower than sale price"
-        );
 
         //
         // 2.
         // Process sale
         //
         address seller = _NFCHAMP_CONTRACT.ownerOf(tokenId);
-        uint256 sellerTokenWeiShare;
-        uint256 marketplaceFeesTokenWeiShare;
-        (sellerTokenWeiShare, marketplaceFeesTokenWeiShare) = computeSaleShares(
-            salePrice
+        (
+            uint256 sellerShare,
+            uint256 marketplaceSellFeeShare,
+            uint256 marketplaceBuyFeeShare,
+            uint256 marketplaceBurnFeeShare
+        ) = computeSaleShares(salePrice);
+
+        require(
+            _CHAMP_TOKEN_CONTRACT.allowance(msg.sender, address(this)) >=
+                salePrice + marketplaceBuyFeeShare,
+            "ChampMarketplace: Allowance is lower than buyer sale price"
         );
 
         //
         // 3.
         // Execute sale
         //
         delete _sales[tokenId];
+
         _NFCHAMP_CONTRACT.safeTransferFrom(seller, nftReceiver, tokenId);
-        _CHAMP_TOKEN_CONTRACT.safeTransferFrom(
-            msg.sender,
-            seller,
-            sellerTokenWeiShare
-        );
-        if (marketplaceFeesTokenWeiShare > 0) {
+        _CHAMP_TOKEN_CONTRACT.safeTransferFrom(msg.sender, seller, sellerShare);
+
+        uint256 marketplaceFeesShare = marketplaceSellFeeShare +
+            marketplaceBuyFeeShare;
```

Comment:
P4: Same here, + we could compute it only once at the beginning

Reply:
Not sure I understand 🤔 in this situation we're calculating the total amount that the fee receiver will receive: **sell fee + buy fee**. This is different than the buyer price: **full listing price + buy fee**.